### PR TITLE
fix logic for kubernetes version

### LINKED ISSUE #7582: Fix Logic to fetch kubernetes version for cluster
#### Current Behavior

[getKubernetesVersion()](https://github.com/meshery/meshery/blob/33a9315fbb21543d444a920b64933644be74f807/ui/components/MeshConfigComponent.js#L253-L265) uses the registered workload in cluster to determine the version of kubernetes of cluster which is totally wrong, given:
1. There are static components whose versions may not match to the active kubernetes cluster
2. hardcoded with 0th index, that doesn't derive confidence of authnencity of data
3. using oam, which has been terminated.
This function is used here in [configTable](https://github.com/meshery/meshery/blob/33a9315fbb21543d444a920b64933644be74f807/ui/components/MeshConfigComponent.js#L155)

#### Expected Behavior

The kubernets version should be detected from the cluster and not from the version of registered components.
#### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by MUzairS15**
**Notes for Reviewers**

This PR fixes #7582 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/models/k8s_context.go [hunk 1]
```diff
@@ -291,6 +306,16 @@ func (kc K8sContext) GenerateKubeHandler() (*kubernetes.Client, error) {
 	return kubernetes.New(cfg)
 }
 
+func (kc *K8sContext) AssignVersion(handler *kubernetes.Client) error {
```
Comment (Abhishek-kumar09) [inline server/models/k8s_context.go:309]: Is this function being used outside of package, if not, we can change it to be available on current scope only.
```suggestion
func (kc *K8sContext) assignVersion(handler *kubernetes.Client) error {
```

**Review by theBeginner86 — APPROVED** [review]
LGTM

**Comment by theBeginner86** [comment]
@MUzairS15 
To ensure consistency, can we merge this PR?
We already merged the corresponding changes in remote provider.

**Comment by MUzairS15** [comment]
Yes ready to merge from long.

**Pull request merged**